crates/nu-command/src/filters/last.rs

```diff
@@ -75,18 +77,29 @@ impl Command for Last {
         let span = call.head;
 
         let rows: Option<i64> = call.opt(engine_state, stack, 0)?;
-        let v: Vec<_> = input.into_iter().collect();
-        let vlen: i64 = v.len() as i64;
-        let beginning_rows_to_skip = rows_to_skip(vlen, rows);
+        let to_keep = rows.unwrap_or(1) as usize;
```

Comment:
This doesn't guard against potential negative numbers as input and would cause and underflow.